cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1010,21 +1013,30 @@ private QueryBuilder createMainQuery(SearchRequest request) {
     // all terms must occur in the text field as selected by the scope in the search request.
     String textField = request.getScope() == null ? TEXT_FIELD : request.getScope().name().toLowerCase();
 
-    // split the query into its terms and iterate over all terms
-    Iterable<String> terms = Splitter.on(SPACE_SEPARATOR_PATTERN)
-      .omitEmptyStrings().trimResults().split(request.getQuery());
-    List<QueryBuilder> termQueries = new ArrayList<>();
-    for (String term : terms) {
-      termQueries.add(createTermQuery(term, textField, request));
+    // split the query into its parsed terms and create corresponding QueryBuilders for each
+    Map<QueryBuilder, QueryTerm> termQueries = new HashMap<>();
+    for (QueryTerm queryTerm : QueryParser.parse(request.getQuery())) {
+      termQueries.put(createTermQuery(queryTerm.getTerm(), textField, request), queryTerm);
     }
     if (termQueries.isEmpty()) {
       return QueryBuilders.matchAllQuery();
     }
-    if (termQueries.size() == 1) {
```

Comment:
doesn't this still apply? If there is only one term, we don't need to construct a bool query